src/modules/hero/hero.component.ts

```diff
@@ -15,5 +16,43 @@ export class StacheHeroComponent {
     return this._backgroundImageUrl;
   }
 
+  @Input()
+  set overlayOpacity(opacityPercentage: string) {
+    let validOpacityInput = opacityPercentage.replace(/[^\d.-]/g, '');
+    let validOpacityAsString = this._parseOpacityValueToDecimal(validOpacityInput);
+    this._overlayOpacity = validOpacityAsString;
+  }
+
+  get overlayOpacity(): string {
+    return this._overlayOpacity;
+  }
+
   private _backgroundImageUrl = 'none';
+  private _overlayOpacity = '0.6';
+
+  private _parseOpacityValueToDecimal(opacityValue: string): string {
+    let percentageCandiate = parseInt(opacityValue, 10);
+    if (opacityValue.includes('.')) {
+      percentageCandiate = Math.round(parseFloat(opacityValue) * 100) / 100;
+    }
+    let validDecimal = this._validateOpacityDecimal(percentageCandiate);
+    return String(validDecimal);
+  }
+
+  private _validateOpacityDecimal(value: number): number {
```

Comment:
Angular [recommends](https://angular.io/styleguide#!#03-04) NO underscores for private properties and methods. In the case of a getter/setter, an underscore is valid.